components/Buttons/tests/unit/MDCButtonTests.m

```diff
@@ -590,6 +590,77 @@ - (void)testTitleFontForStateWithLegacyDynamicTypeRestoresUserFonts {
                 self.button.titleLabel.font);
 }
 
+- (void)testTitleFontForStateDisabledAfterSettingFontsPreventsFontChange {
+  // Given
+  UIFont *normalFont = [UIFont systemFontOfSize:10];
+  UIFont *selectedFont = [UIFont systemFontOfSize:40];
+  UIFont *directlyAssignedFont = [UIFont systemFontOfSize:25];
+  [self.button setTitleFont:normalFont forState:UIControlStateNormal];
+  [self.button setTitleFont:selectedFont forState:UIControlStateSelected];
+
+  // When
+  self.button.enableTitleFontForState = NO;
+  self.button.titleLabel.font = directlyAssignedFont;
+  self.button.selected = YES;
+
+  // Then
+  XCTAssertEqualObjects(self.button.titleLabel.font, directlyAssignedFont);
+}
+
+- (void)testTitleFontForStateDisabledBeforeSettingFontsPreventsFontChange {
```

Comment:
s/disabled/selected/
looks like we are setting the selected font not the disabled font.

and above.

Reply:
The name of the test reflects disabling `titleFontForState`, not the button state. Can you recommend a clearer name?

Reply:
hrm struggling here now that I understand what you are testing...
best I could come up with:
testDisabledTitleFontForStateFlagBeforeSettingFontForStateShouldNotChangeFonts

might be worth a comment to explain that it is a test for the flag.

Reply:
Naming is hard. I just added comments to all of the new tests.